infrastructure/io/src/main/java/tech/pegasys/teku/infrastructure/io/SyncDataAccessor.java

```diff
@@ -45,13 +81,27 @@
 
   /**
    * Writes data to the specified path, ensuring both the file content and metadata is immediately
-   * flushed to hardware storage. If the file exists it is overwritten, otherwise it is created.
+   * flushed to hardware storage. If the filesystem support atomic operation it creates a temporary
+   * file before writing the actual file. If the file exists it is overwritten, otherwise it is
+   * created.
    *
    * @param path the path to write to
    * @param data the data to write
    * @exception IOException if an IO error occurs while writing
    */
   public void syncedWrite(final Path path, final Bytes data) throws IOException {
+    if (atomicFileMoveSupport) {
+      try {
+        atomicSyncedWrite(path, data);
+      } catch (AtomicMoveNotSupportedException e) {
+        nonAtomicSyncedWrite(path, data);
+      }
```

Comment:
I wouldn't catch this - if we originally detected that atomic moves were supported and they suddenly aren't then that's a sign that something very strange is going on.  Plus doing this fall back every time we sign will degrade performance as we're writing and fsyncing the change to disk twice so we want the user to be alerted.

So if we get here and the atomic move fails, just let the exception throw and let signing fail.